vnext/ReactUWP/Views/SIPEventHandler.cpp

```diff
@@ -20,38 +21,85 @@ namespace react {
-            winrt::Rect finalRect = winrt::RectHelper::Empty();
-            winrt::IVectorView<winrt::CoreInputViewOcclusion> occlusions = e.Occlusions();
-            for (uint32_t i = 0; i < occlusions.Size(); i++) {
-              winrt::CoreInputViewOcclusion occlusion = occlusions.GetAt(i);
-              if (occlusion.OcclusionKind() == winrt::CoreInputViewOcclusionKind::Docked) {
-                finalRect = winrt::RectHelper::Union(finalRect, occlusion.OccludingRect());
+            bool wasShowing = m_isShowing;
+            m_isShowing = !IsOcclusionsEmpty(e.Occlusions());
+            if (wasShowing != m_isShowing && m_fireKeyboradEvents) {
+              if (!m_isShowing) {
+                folly::dynamic params = folly::dynamic::object("screenY", 0)("screenX", 0)("width", 0)("height", 0);
+                SendEvent("keyboardDidHide", std::move(params));
+              } else {
+                folly::dynamic params = folly::dynamic::object(
+                    "endCoordinates",
+                    folly::dynamic::object("screenY", m_finalRect.Y)("screenX", m_finalRect.X)(
+                        "width", m_finalRect.Width)("height", m_finalRect.Height));
+                SendEvent("keyboardDidShow", std::move(params));
               }
             }
-
-            if (winrt::RectHelper::GetIsEmpty(finalRect)) {
-              folly::dynamic params = folly::dynamic::object("screenY", 0)("screenX", 0)("width", 0)("height", 0);
-              SendEvent("keyboardDidHide", std::move(params));
-            } else {
-              folly::dynamic params = folly::dynamic::object(
-                  "endCoordinates",
-                  folly::dynamic::object("screenY", finalRect.Y)("screenX", finalRect.X)("width", finalRect.Width)(
-                      "height", finalRect.Height));
-              SendEvent("keyboardDidShow", std::move(params));
-            }
           }
         });
   }
 }
 
-SIPEventHandler::~SIPEventHandler() {
-  m_occlusionsChanged_revoker = {};
+bool SIPEventHandler::TryShow() {
+  if (IsRS5OrHigher() && m_coreInputView) {
+    if (!m_isShowing) { // CoreInputView.TryShow is only avaliable after RS5
+      return m_coreInputView.TryShow();
+    }
+    return true;
+  }
+  return false;
+}
+
+bool SIPEventHandler::TryHide() {
+  if (IsRS5OrHigher() && m_coreInputView) {
+    if (m_isShowing) { // CoreInputView.TryHide is only avaliable after RS5
+      return m_coreInputView.TryHide();
+    }
+    return true;
+  }
+  return false;
+}
+
+bool SIPEventHandler::IsOcclusionsEmpty(winrt::IVectorView<winrt::CoreInputViewOcclusion> occlusions) {
+  m_finalRect = winrt::RectHelper::Empty();
+  if (occlusions) {
+    for (uint32_t i = 0; i < occlusions.Size(); i++) {
+      winrt::CoreInputViewOcclusion occlusion = occlusions.GetAt(i);
+      if (occlusion.OcclusionKind() == winrt::CoreInputViewOcclusionKind::Docked) {
+        m_finalRect = winrt::RectHelper::Union(m_finalRect, occlusion.OccludingRect());
+      }
```

Comment:
nit: cppwinrt supports for like this
for (auto const& item: occlusions) {
}

or 
for (auto item: occlusions) {
}
I don't know if the first one will work or not